libs/input-mapping/src/Table/Strategy/AbstractWorkspaceStrategy.php

```diff
@@ -183,4 +189,174 @@ public function handleExports(array $exports, bool $preserve): array
+     * @return WorkspaceTableLoadInstruction[]
+     */
+    public function prepareTableLoadsToWorkspace(array $tables): array
+    {
+        $instructions = [];
+
+        foreach ($tables as $table) {
+            $loadOptions = $table->getStorageApiLoadOptions($this->tablesState);
+
+            // Validate that table can be loaded to this workspace type
+            LoadTypeDecider::checkViableLoadMethod(
+                $table->getTableInfo(),
+                $this->getWorkspaceType(),
+                $loadOptions,
+                $this->clientWrapper->getToken()->getProjectId(),
+            );
+
+            // Determine optimal load method from Table Storage to Workspace
+            if (LoadTypeDecider::canClone($table->getTableInfo(), $this->getWorkspaceType(), $loadOptions)) {
+                $this->logger->info(sprintf('Table "%s" will be cloned.', $table->getSource()));
+                $instructions[] = new WorkspaceTableLoadInstruction(
+                    WorkspaceLoadType::CLONE,
+                    $table,
+                    null,
+                );
+            } elseif (LoadTypeDecider::canUseView($table->getTableInfo(), $this->getWorkspaceType())) {
+                $this->logger->info(sprintf('Table "%s" will be created as view.', $table->getSource()));
+                $instructions[] = new WorkspaceTableLoadInstruction(
+                    WorkspaceLoadType::VIEW,
+                    $table,
+                    $loadOptions,
+                );
+            } else {
+                $this->logger->info(sprintf('Table "%s" will be copied.', $table->getSource()));
+                $instructions[] = new WorkspaceTableLoadInstruction(
+                    WorkspaceLoadType::COPY,
+                    $table,
+                    $loadOptions,
+                );
+            }
+        }
+
+        return $instructions;
+    }
+
+    /**
+     * Phase 2: Execute - Submit async jobs to load tables from Table Storage into Workspace
+     * CLEAN jobs must complete before other jobs are submitted
+     */
+    public function executeTableLoadsToWorkspace(WorkspaceLoadPlan $plan): WorkspaceLoadQueue
+    {
+        $jobs = [];
+        $workspaces = $this->createWorkspaces();
+        $workspaceId = (int) $this->dataStorage->getWorkspaceId();
+
+        // Step 1: Clean workspace if needed - MUST complete before other operations
+        if (!$plan->preserve) {
+            $this->logger->info('Cleaning workspace before loading tables.');
+            $cleanJobId = $workspaces->queueWorkspaceCloneInto($workspaceId, [
+                'input' => [], // workspace will be only cleaned
+                'preserve' => 0,
+            ]);
+
+            // Wait for clean job to complete before proceeding
+            $this->clientWrapper->getBranchClient()->handleAsyncTasks([$cleanJobId]);
+
+            // Don't add CLEAN job to queue since it's already completed
+        }
```

Comment:
Tohle je odlisnost od puvodni implementace.

Tam to bylo synchronne za sebou. Clone a pak Copy. Pokud nemel byt preserve, tak prvni job i promazal workspace.

Tady ale chceme aby to bylo async. A zafrontovaly se oba joby.
Takze cleanup se musi vyresit predtim. Proto tenhle job. Zaroven se ceka na nej, nez se dokonci

Reply:
No to je prekerní tohle, vůbec se mě to neliíbí, protože to je zase hroznej overhead s jobama, ale jako lepší řešní nemám (pochopitelně bez toho, aby to podporovalo connection a týhle věci jsme se úplně zbavili). 

Napadá mě jedině ten job založit a podívat se jestli se switchnul na processing a pak teprv založit ten druhej, čimž by se zařídilo, že jdou po sobě, jenže by to současně znamenalo, že když se zafrontují, tak se to celý zasekne a vytimoutuje, což je naprd.
Jenže stejně tak se to celý zafrontuje a vytimoutuje i když se bude čekat na ten cleanup job.